lib/schema.js

```diff
@@ -636,18 +636,23 @@ Schema.prototype.execPreHandlers = function(event, mdlInst, callback) {
     callback(null);
     return;
   }
-  
+
   var preHandlers = this.preHandlers[event];
   var i = 0;
-  var doNext = function _doNextPreHandler() {
+  var doNext = function _doNextPreHandler(err) {
+    // If any one of the pre-handlers fails, fail the whole thing.
+    // This allows people to plug in additional validation.
+    if(err) { return callback(err); }
+
     if (i === preHandlers.length) {
       callback(null);
       return;
     }
 
     var curI = i;
     i++;
-    preHandlers[curI].call(mdlInst, doNext);
+
+    preHandlers[curI].call(mdlInst, mdlInst, doNext);
```

Comment:
Note the change to call; previously, mdlInst was set to the "this" for the function, meaning its first argument was a callback in error.